answer.js

```diff
@@ -1,17 +1,419 @@
+// 	name: String,
+// 	email: {
+// 		lowercase: true,
+// 		type: String
+// 	},
+// 	age: {
+// 		type: Number,
+// 		default: 0
+// 	}
+// });
+// // Q3 Answer here...
+// var userSchema = new Schema({
+// 	name: String,
+// 	email: {
+// 		lowercase: true,
+// 		type: String
+// 	},
+// 	age: {
+// 		type: Number,
+// 		default: 0
+// 	},
+// 	password: {
+// 		type: String,
+// 		minlength: 5
+// 	},
+// 	createdAt: {
+// 		type: Date,
+// 		default: new Date()
+// 	}
+// });
 
-// Q3 Answer here...
+// // Q4 Answer here...
+// var addressSchema = new Schema({
+// 	village: String,
+// 	city: String,
+// 	state: String,
+// 	pin: Number,
+// 	user: Schema.Types.ObjectId
+// });
 
-// Q4 Answer here...
+// // Q5 Answer here...
+// var addressSchema = new Schema({
+// 	village: String,
+// 	city: {
+// 		type: String,
+// 		required: true
+// 	},
+// 	state: {
+// 		type: String,
+// 		required: true
+// 	},
+// 	pin: Number,
+// 	user: Schema.Types.ObjectId
+// });
 
-// Q5 Answer here...
+// // Q6 Answer here...
+// var userSchema = new Schema({
+// 	name: String,
+// 	email: {
+// 		lowercase: true,
+// 		type: String
+// 	},
+// 	age: {
+// 		type: Number,
+// 		default: 0
+// 	},
+// 	password: {
+// 		type: String,
+// 		minlength: 5
+// 	},
+// 	createdAt: {
+// 		type: Date,
+// 		default: new Date()
+// 	},
+// 	favourites: {
+// 		type: [ String ],
+// 		required: true
+// 	}
+// });
 
-// Q6 Answer here...
+// // Q7 Answer here...
 
-// Q7 Answer here...
+// var addressSchema = new Schema(
+// 	{
+// 		village: String,
+// 		city: {
+// 			type: String,
+// 			required: true
+// 		},
+// 		state: {
+// 			type: String,
+// 			required: true
+// 		},
+// 		pin: Number,
+// 		user: Schema.Types.ObjectId
+// 	},
+// 	{ timestamps: true }
+// );
+// var userSchema = new Schema(
+// 	{
+// 		name: String,
+// 		email: {
+// 			lowercase: true,
+// 			type: String
+// 		},
+// 		age: {
+// 			type: Number,
+// 			default: 0
+// 		},
+// 		password: {
+// 			type: String,
+// 			minlength: 5
+// 		},
+// 		createdAt: {
+// 			type: Date,
+// 			default: new Date()
+// 		},
+// 		favourites: {
+// 			type: [ String ],
+// 			required: true
+// 		}
+// 	},
+// 	{ timestamps: true }
+// );
 
-// Q8 Answer here...
\ No newline at end of file
+// // Q8 Answer here...
+// module.exports = mongoose.model('User', userSchema);
+// module.exports = mongoose.model('Address', addressSchema);
+
+// Q9 Answer here..
+// let express = require('express');
+// let mongoose = require('mongoose');
+// let User = require('./models/users');
+// let app = express();
+
+// var PORT = process.env.PORT || 3000;
+
+// // connecting to mongodb database
+
+// mongoose.connect(
+// 	'mongodb://localhost/mongoose',
+// 	{
+// 		useNewUrlParser: true,
+// 		useUnifiedTopology: true,
+// 		useCreateIndex: true
+// 	},
+// 	(err) => {
+// 		console.log('Connected', err ? err : true);
+// 	}
+// );
+
+// //instantiate express app
+// app.use(express.json());
```

Comment:
After instantiating express app, we have stacks of middlewares, which starts with `app.use()`